### PR TITLE
feat(hx-file-icon): new element styles added for hx-file-icon

### PR DESCRIPTION
**Pull request by cathysiller**
JIRA: SURF-1103

### LGTM's
- [x] Dev LGTM
- [x] Design LGTM
- [x] Zoom LGTM

## hx-file-icon
file-icons-helixui

icons-helixui

hx-file-icon-helixui

### CODE REVIEW

File: docs/elements/hx-file-icon/index.html [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    </p>
+
+    <hx-dl class="hxBox-md metadata">
+      <hx-def>
+        <hx-dt>Permitted Parents</hx-dt>
+        <hx-dd>any element that accepts embeded content</hx-dd>
+      </hx-def>
+      <hx-def>
+        <hx-dt>Permitted Children</hx-dt>
+        <hx-dd>none</hx-dd>
```
Comment (CITguy) [inline docs/elements/hx-file-icon/index.html]: (phrasing content) file extension

File: docs/elements/hx-file-icon/index.html [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  <section>
+    <p>
+      The custom <code>{{page.title}}</code> element renders a "file" icon with 
+      a customizable image and extension.
+    </p>
+
+    <hx-dl class="hxBox-md metadata">
+      <hx-def>
+        <hx-dt>Permitted Parents</hx-dt>
+        <hx-dd>any element that accepts embeded content</hx-dd>
```
Comment (CITguy) [inline docs/elements/hx-file-icon/index.html]: s/embeded/flow/

File: docs/elements/hx-file-icon/index.html [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        <hx-dt>Events</hx-dt>
+        <hx-dd><em>none</em></hx-dd>
+      </hx-def>
+    </hx-dl>
+  </section>
+{% endblock %}
+
+{% block attributes %}
+  <dl>
+    <dt>type</dt>
```
Comment (CITguy) [inline docs/elements/hx-file-icon/index.html]: `type {String}`

File: docs/elements/hx-icon/index.html [hunk 4]
```diff
@@ -8,7 +8,7 @@
 {% block content %}
```
Comment (CITguy) [inline docs/elements/hx-icon/index.html]: Add links between `<hx-icon>` and `<hx-file-icon>` element docs.

File: docs/elements/hx-file-icon/index.html [hunk 5]
```diff
@@ -0,0 +1,39 @@
+---
+title: <hx-file-icon>
+also:
+  components/icons: Icons
```
Comment (CITguy) [inline docs/elements/hx-file-icon/index.html]: Add links between <hx-icon> and <hx-file-icon> element docs.

File: src/helix-ui/elements/HXFileIconElement.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+            this._elIcon.type = newVal;
+        }
+    }//$onAttributeChange
+
+    // GETTERS
+    get type () {
+        return this.getAttribute('type');
+    }
+
+    get _elIcon () {
```
Comment (CITguy) [inline src/helix-ui/elements/HXFileIconElement.js]: move below the `type` getter/setter

File: src/helix-ui/elements/HXFileIconElement.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
+import { HXElement } from './HXElement';
+import shadowStyles from './HXFileIconElement.less';
+import shadowMarkup from './HXFileIconElement.html';
+
+export class HXFileIconElement extends HXElement {
```
Comment (CITguy) [inline src/helix-ui/elements/HXFileIconElement.js]: Please add basic jsdoc comment.

File: src/helix-ui/elements/HXFileIconElement.less [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+  }
+
+  #overlay {
+    line-height: 1;
+    position: absolute;
+    text-align: center;
+    top: 0.5rem;
+    width: 100%;
+
+    hx-icon {
```
Comment (CITguy) [inline src/helix-ui/elements/HXFileIconElement.less]: `#icon`

File: src/helix-ui/elements/HXFileIconElement.less [hunk 9]
```diff
@@ -0,0 +1,30 @@
+@import './HXElement';
+@import 'elements/hx-icon';
+
+#wrapper {
+  position: relative;
+
+  #base {
```
Comment (CITguy) [inline src/helix-ui/elements/HXFileIconElement.less]: remove nesting for all of the id selector blocks.

File: docs/elements/hx-icon/index.html [hunk 10]
```diff
@@ -3,23 +3,24 @@
...
     <hx-dl class="hxBox-md metadata">
       <hx-def>
         <hx-dt>Permitted Parents</hx-dt>
-        <hx-dd>any element that accepts embeded content</hx-dd>
+        <hx-dd>any element that accepts embedded content</hx-dd>
       </hx-def>
       <hx-def>
         <hx-dt>Permitted Children</hx-dt>
-        <hx-dd>none</hx-dd>
+        <hx-dd>(phrasing content) file extension</hx-dd>
```
Comment (CITguy) [inline docs/elements/hx-icon/index.html]: Please revert this change. Looks like it was updated by mistake. The `<hx-icon>` element doesn't permit any children.

File: docs/elements/hx-file-icon/index.html [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+    </p>
+
+    <hx-dl class="hxBox-md metadata">
+      <hx-def>
+        <hx-dt>Permitted Parents</hx-dt>
+        <hx-dd>any element that accepts flow content</hx-dd>
+      </hx-def>
+      <hx-def>
+        <hx-dt>Permitted Children</hx-dt>
+        <hx-dd>(phrasing content) file extension</hx-dd>
```
Comment (CITguy) [inline docs/elements/hx-file-icon/index.html]: `phrasing content (specifically, an optional file extension)`

**Comment by araiford** [comment]
@cathysiller -- awesome!

One tiny nit pick...  Is there any way to bump the extension down 1 pixel?  The extensions are crowding the icon a little.

Here's a screenshot with the 1px bump:

image

Aside from that this looks great!

**Comment by cathysiller** [comment]
hey @araiford !! yes does look a little crowded, added `margin-top: 1px` to the `#ext`!!

**Comment by araiford** [comment]
:shipit:
